catalyst_voices/packages/internal/catalyst_voices_shared/lib/src/range/range.dart

```diff
@@ -0,0 +1,13 @@
+class Range<T extends num> {
+  final T min;
+  final T max;
+
+  const Range({required this.min, required this.max});
+
+  static Range<int>? createIntRange({int? min, int? max}) {
+    if (min == null || max == null) {
+      return null;
+    }
+    return Range(min: min, max: max);
+  }
```

Comment:
How about this approach? You won't need to add more methods for more types, just handle it once generically:

```suggestion
  static Range<T>? optionalRangeOf<T extends num>({T? min, T? max}) {
    if (min == null || max == null) {
      return null;
    }
    return Range(min: min, max: max);
  }
```